functions/index.js

```diff
@@ -75,19 +92,33 @@ exports.pullDataFromSftp= functions.https.onRequest((request, response) => {
         //
         //        Name                Year       Month   Day      Hour
         //        res[1]              res[2]     res[3]  res[4]   res[5]
-        fRegEx = /(gm_clients_served_)(\d\d\d\d)-(\d?\d)-(\d?\d)-?(\d?\d)?/;
+
         // Applies the regex pattern to our filename and stores result in fnParts (file name parts)
-        fnParts = fRegEx.exec(fileName);
+        switch (clientOrGoalCSV) {
+          case 'client':
+            fRegEx_gm_clients = /(gm_clients_served_)(\d\d\d\d)-(\d?\d)-(\d?\d)-?(\d?\d)?/;
+            fnParts = fRegEx_gm_clients.exec(fileName);
+            break;
+          case 'goal':
+            fRegEx_gm_goals = /(gm_goals_details_)(\d\d\d\d)-(\d?\d)-(\d?\d)-?(\d?\d)?/;        
+            fnParts = fRegEx_gm_goals.exec(fileName);
+            break;
+          default:
+            throw ("Must provide argument <client|goal>");
```

Comment:
If the error snippet above works, you can reuse it in the error cases here and elsewhere throughout the file.